src/endpoints/user.py

```diff
@@ -0,0 +1,298 @@
+import logging
+from typing import List
+
+from fastapi import APIRouter, Depends, HTTPException, Path, Query
+from sqlalchemy import delete, select
+from sqlalchemy.orm import Session
+from starlette import status
+
+from ..models.database import get_db
+from ..models.user import User
+from ..schemas.update_user import UpdateUserSchema
+from ..schemas.user import UserSchema
+
+from .auth import (  # isort: skip
+    get_current_librarian,  # isort: skip
+    get_current_user,  # isort: skip
+    get_password_hash,  # isort: skip
+    verify_password,  # isort: skip
+)  # isort: skip
+
+router = APIRouter(
+    prefix="/user", tags=["user"], responses={401: {"user": "Not authorized"}}
+)
+
+
+@router.get("", status_code=status.HTTP_200_OK, response_model=None)
+async def filter_user(
+    email: str | None = None,
+    username: str | None = None,
+    first_name: str | None = None,
+    last_name: str | None = None,
+    contact_number: str | None = None,
+    address: str | None = None,
+    librarian: dict = Depends(get_current_librarian),
+    db: Session = Depends(get_db),
+) -> None:
+    """
+    Filters the users list based on param provided. If None given then it will
+    return the complete list.
+    """
+    params = {
+        "email": email,
+        "username": username,
+        "first_name": first_name,
+        "last_name": last_name,
+        "contact_number": contact_number,
+        "address": address,
+    }
+    filters = {key: value for key, value in params.items() if value}
+    try:
+        users = db.query(User).filter_by(**filters).all()
+        return users
+    except Exception:
+        logging.exception(f"Exception occured -- {__name__}.filter_user")
+        raise db_not_available()
+
+
+@router.get("/", status_code=status.HTTP_200_OK, response_model=None)
```

Comment:
Is this endpoint really required since the filter one returns all users when no parameter is required?